core/src/main/java/org/mule/runtime/core/internal/processor/strategy/ProactorStreamEmitterProcessingStrategyFactory.java

```diff
@@ -128,14 +129,14 @@ public ProactorStreamEmitterProcessingStrategy(Supplier<Scheduler> ringBufferSch
     @Override
     public Sink createSink(FlowConstruct flowConstruct, ReactiveProcessor function) {
       final long shutdownTimeout = flowConstruct.getMuleContext().getConfiguration().getShutdownTimeout();
+      final int sinksCount = maxConcurrency < CORES ? maxConcurrency : CORES;
+      final int sinkBufferSize = bufferSize / sinksCount;
       List<ReactorSink<CoreEvent>> sinks = new ArrayList<>();
-      int concurrency = maxConcurrency < subscribers ? maxConcurrency : subscribers;
-      reactor.core.scheduler.Scheduler scheduler = fromExecutorService(decorateScheduler(getCpuLightScheduler()));
-      for (int i = 0; i < concurrency; i++) {
+
+      for (int i = 0; i < sinksCount; i++) {
         Latch completionLatch = new Latch();
-        EmitterProcessor<CoreEvent> processor = EmitterProcessor.create(bufferSize / concurrency);
-        processor.publishOn(scheduler).doOnSubscribe(subscription -> currentThread().setContextClassLoader(executionClassloader))
-            .transform(function).subscribe(null, e -> completionLatch.release(), () -> completionLatch.release());
+        EmitterProcessor<CoreEvent> processor = EmitterProcessor.create(sinkBufferSize);
+        processor.transform(function).subscribe(null, e -> completionLatch.release(), () -> completionLatch.release());
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Replace this lambda with a method reference. ([squid:S1612](https://sonarqube.kbuild.msap.io/coding_rules#rule_key=squid%3AS1612))

[See it in SonarQube](https://sonarqube.kbuild.msap.io/project/issues?id=mule-runtime-mule&pullRequest=7657&issues=AWjJeTBLotIXnmsUTyuM&open=AWjJeTBLotIXnmsUTyuM)

[ISSUE_KEY]: AWjJeTBLotIXnmsUTyuM